### PR TITLE
Module for attribute virt:resource:assignedGroups

### PR DESCRIPTION
**Pull request by stavamichal**
- module will compute list of group names assigned to the resource
 (optionally also with all their subgroups)

### CODE REVIEW

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_virt_assignedGroups.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	 * @return computed group name
+	 */
+	private String computeGroupName(Vo vo, Group group) {
+		return vo.getShortName() + ":" + group.getName();
+	}
+
+	@Override
+	public List<String> getStrongDependencies() {
+		List<String> strongDependencies = new ArrayList<>();
+		strongDependencies.add(AttributesManager.NS_VO_ATTR_CORE + ":shortName");
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_virt_assignedGroups.java:90]: Why ? We retrieve value from the object ``Vo`` and do not call ``getAttribute()`` on such core attribute.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_virt_assignedGroups.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+		for(Group assignedGroup: assignedGroups) {
+			allUniqueGroupNames.add(computeGroupName(vo, assignedGroup));
+			if(isAssignedWithSubgroups) {
+				for(Group subgroup: sess.getPerunBl().getGroupsManagerBl().getAllSubGroups(sess, assignedGroup)) {
+					allUniqueGroupNames.add(computeGroupName(vo, subgroup));
+				}
+			}
+		}
+
+		attribute.setValue(new ArrayList(allUniqueGroupNames));
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_virt_assignedGroups.java:71]: Maybe we should sort list of group names to ensure consistent value is returned on each request. I think we can use ``SortedSet`` to do it on the fly.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_virt_assignedGroups.java (hunk 1 above)
Comment (stavamichal) [reply perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_virt_assignedGroups.java]: Copy&Paste mistake. Fixed. :)

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_virt_assignedGroups.java (hunk 2 above)
Comment (stavamichal) [reply perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_virt_assignedGroups.java]: Added.

**Pull request merged**